mesonbuild/compilers/c.py

```diff
@@ -704,10 +706,12 @@ def has_function(self, funcname, prefix, env, extra_args=None, dependencies=None
              * We would always check for this, but we get false positives on
              * MSYS2 if we do. Their toolchain is broken, but we can at least
              * give them a workaround. */
-            #if {no_includes:d}
+            #if {no_includes:d} && !{func_is_builtin:d}
                 __builtin_{func};
+            #elif defined(__stub_{func}) || defined(_stub__{func})
```

Comment:
Do we need a comment similar to the one near stubs_fail to make it easier to understand. 
Do we need to make sure that limits.h is included? Because the comment above implies it is needed for this to work.

Reply:
Fixed.